dvc/repo/__init__.py

```diff
@@ -189,14 +189,44 @@ def check_modified_graph(self, new_stages):
         #
         # [1] https://github.com/iterative/dvc/issues/2671
         if not getattr(self, "_skip_graph_checks", False):
-            self._collect_graph(self.stages + new_stages)
+            self._collect_graph((old_stages or self.stages) + new_stages)
+
+    def collect_for_pipelines(
+        self, path=None, name=None, recursive=False, graph=None
+    ):
+        from .. import dvcfile
+        import networkx as nx
+
+        if not path:
+            return list(graph) if graph else self.pipeline_stages
+
+        dvcfile = dvcfile.Dvcfile(self, path)
+        if dvcfile.exists():
+            path = os.path.abspath(path)
+
+            if recursive and os.path.isdir(path):
+                stages = nx.dfs_postorder_nodes(graph or self.pipeline_graph)
+                return [
+                    stage for stage in stages if path_isin(stage.path, path)
+                ]
+            return [dvcfile.load_one(name)]
+
+        from dvc.stage import PipelineStage
```

Comment:
Let's keep the imports orginised, no need to throw them around all over. Also, need to preserve order: system imports, 3rd party libs, our imports.